src/platform/packages/shared/content-management/table_list_view_table/src/query_client.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { defaultContext, QueryClient, QueryClientProvider } from '@tanstack/react-query';
+
+const defaultTableQueryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false } },
+});
+
+/**
+ * Attempts to reuse existing query client otherwise fallbacks to a default one.
+ */
+export function MaybeQueryClientProvider({ children }: React.PropsWithChildren) {
+  const client = React.useContext(defaultContext);
```

Comment:
I reworked a bit query client provider in table list view to allow re-using existing QueryClientProvider. This allowed to share cache for favorites between table list view and dashboard page